lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -983,4 +1035,41 @@ namespace Js
         }
         return slotIndex;
     }
+
+    // static
+    Var SourceTextModuleRecord::ResolveOrRejectDynamicImportPromise(bool toResolve, Var value, ScriptContext *scriptContext, SourceTextModuleRecord *moduleRecord)
+    {
+        JavascriptPromiseCapability *promiseCapability = nullptr;
+        if (moduleRecord != nullptr)
+        {
+            promiseCapability = moduleRecord->GetPromiseCapability();
+        }
+
+        if (promiseCapability == nullptr)
+        {
+            promiseCapability = JavascriptPromise::NewPromiseCapability(scriptContext->GetLibrary()->GetPromiseConstructor(), scriptContext);
+        }
+
+        Var resolveOrRejectVar = toResolve ? promiseCapability->GetResolve() : promiseCapability->GetReject();
```

Comment:
So `toResolve` indicates the promise should be resolved instead of rejected? Nitpicky, but I think I would prefer this to be named `isResolve`, `isReject`, `success`, or something along those lines.